pkg/trace/pb: add documentation comments to span.proto

dgoffredo opened the pull request:
I was trying to figure out in which units a span's `start` and `duration` are expressed.  After looking at some code, I found that the units are nanoseconds.  I'd like to document that in the `.proto` into which tracer payloads are decoded.

While I'm at it, might as well document the other fields as well.

dgoffredo committed:
add documentation comments to pkg/trace/pb/span.proto

dgoffredo committed:
figured out what meta_struct is for

dgoffredo commented:
https://app.circleci.com/pipelines/github/DataDog/datadog-agent/43407/workflows/68867d50-9e3f-4bf0-81da-438fec4ea82b/jobs/613159

> PR https://github.com/DataDog/datadog-agent/pull/11731 requires a milestone.

Does it?

https://app.circleci.com/pipelines/github/DataDog/datadog-agent/43407/workflows/68867d50-9e3f-4bf0-81da-438fec4ea82b/jobs/613154

> Error: No releasenote was found for this PR. Please add one using 'reno', or apply the label 'changelog/no-changelog' to the PR.

I don't think that comments in a `.proto` file deserve mention in release notes.

https://app.circleci.com/pipelines/github/DataDog/datadog-agent/43407/workflows/68867d50-9e3f-4bf0-81da-438fec4ea82b/jobs/613155

> PR https://github.com/DataDog/datadog-agent/pull/11731 requires team assignment

What is a team, to which should I assign this PR, and how?

ajgajg1134 commented:
> Does it?

Yep! It's just the latest milestone generally (unless the change is needed for a code frozen version). In this case I added the milestone 7.37.0

> What is a team, to which should I assign this PR, and how?

The team you assigned to review `agent-apm` looked good so I just added it as a label to the PR

dgoffredo commented:
> 



> > Does it?
> 
> Yep! It's just the latest milestone generally (unless the change is needed for a code frozen version). In this case I added the milestone 7.37.0
> 
> > What is a team, to which should I assign this PR, and how?
> 
> The team you assigned to review `agent-apm` looked good so I just added it as a label to the PR

Thanks!

dgoffredo commented:
The failing `dd-gitlab/test_macos` check links to a 404 page: https://gitlab.ddbuild.io/datadog/datadog-agent/builds/131098853

gbbr reviewed (approved):
Much appreciated!
